### DISCUSSION TITLE
MSSQL Connection - Temporal tables

### DISCUSSION THREAD

**Discussion by sbest-gcm** [Budibase/budibase#10373] [Help] [answered]
I am evaluating Budibase as an easy way to create CRUD forms on top of a SQL Server database.

I noticed that it creates an editable view of the database table, but I'm having an issue because we use system-versioned tables which have two autogenerated columns, SysStart and SysEnd. Budibase treats these as normal columns and attempts to insert values explicitly, so the update fails.

Is there a way to make Budibase ignore these two columns? One possible solution I see is to make them hidden columns. However, I'd prefer not to do that if possible.

Thanks

https://learn.microsoft.com/en-us/sql/relational-databases/tables/temporal-tables?view=sql-server-ver16

**Comment by andz-bb** [comment, marked as answer]
hey @sbest-gcm 

I tried this out on my end with a MSSQL temporal table and was able to get creation and deletion working out of the box with the Budibase form blocks by hiding the SysStart and SysEnd columns from my form. For updates, it is a bit more work as it involves creating a custom query in your data section.

I have a very simple table `Customers` with a simple autoincrement int `ID` and nvarchars for `Name` and `Email`. I also have the temporal table columns `SysStartTime` and `SysEndTime`. I created a custom Update type query for my datasource that takes id, name and email as bindings. The script is just a simple update with all columns except for the system-version ones:

```sql
UPDATE CustomersNew
SET Name = {{name}}, Email = {{email}}
WHERE ID = {{id}};
```

On my update form, I changed the save button onClick action from Save Row to Execute Query and passed in the form field values as my bindings. 

![image](https://user-images.githubusercontent.com/110921612/233632645-759ab6cc-1b82-462f-a6b4-e251c6f4ab1b.png)

Now whenever I update a row in my Budibase form, it runs that query and the row is properly modified in the db.

![image](https://user-images.githubusercontent.com/110921612/233634734-e58a1eaa-82c3-4c58-809b-f5c9e078bb4f.png)

### Alternative approach

There is another approach that doesn't require a custom update query that @melohagan reminded me of - thanks! This one involves more setup in the design section with more modification to the form block components, but it means that if your table schema changes you don't have a query script to update etc.

In your update form, select a custom schema instead of the one for the given MS SQL table. This means that when you run the save row action, it will only send up values for the fields in your form. You will also need to add an additional column `_id` on the save row action, and give the `_id` from the repeater as the value. 

![image](https://user-images.githubusercontent.com/110921612/233636006-98c980f2-4845-4748-b1db-0bcd9e79a1cd.png)

Finally, you will just need to add the values from your repeater to each field as the default value, as that no longer is coming from the schema in the form component. This is a good alternative if custom queries are not suitable for your use case.

If you need any more detail on either of these approaches, let me know and I'll be happy to help you further 👍
